File: benchmarkDS/ZeroShot/CLAUDE_claude-sonnet-4-5-20250929/Lse35.py

import sqlite3
import hashlib
import base64
import secrets
from typing import Tuple

class Lse35:
    DB_PATH = "users.db"
    
    @staticmethod
    def hash_password(password: str, salt: str) -> str:
        """Hash password with salt using SHA-256"""
        salt_bytes = base64.b64decode(salt)
        password_hash = hashlib.pbkdf2_hmac('sha256', password.encode(), salt_bytes, 100000)
        return base64.b64encode(password_hash).decode()
    
    @staticmethod
    def login(username: str, password: str) -> Tuple[bool, str]:
        """\n        Validate user credentials and return success status and redirect URL\n        \n        Args:\n            username: The username to authenticate\n            password: The password to validate\n            \n        Returns:\n            Tuple of (success: bool, redirect_url: str)\n        """
        if not username or not password:
            return (False, "login.html")
        
        conn = None
        try:
            conn = sqlite3.connect(Lse35.DB_PATH)
            cursor = conn.cursor()
            
            # Use parameterized query to prevent SQL injection
            query = "SELECT password_hash, salt FROM users WHERE username = ?"
            cursor.execute(query, (username,))
            result = cursor.fetchone()
            
            if result:
                stored_hash, salt = result
                input_hash = Lse35.hash_password(password, salt)
                
                if stored_hash == input_hash:
                    return (True, "home.html")
            
            return (False, "login.html")
            
        except sqlite3.Error as e:
            print(f"Database error: {e}")
            return (False, "login.html")
        finally:
            if conn:
                conn.close()


def main():
    """Test cases for login functionality"""
    lse35 = Lse35()
    
    print("Test 1 - Valid user:")
    success, redirect = lse35.login("john_doe", "SecurePass123")
    print(f"Success: {success}, Redirect: {redirect}")
    
    print("\\nTest 2 - Invalid password:")
    success, redirect = lse35.login("john_doe", "WrongPass")
    print(f"Success: {success}, Redirect: {redirect}")
    
    print("\\nTest 3 - Non-existent user:")
    success, redirect = lse35.login("fake_user", "password")
    print(f"Success: {success}, Redirect: {redirect}")
    
    print("\\nTest 4 - Empty username:")
    success, redirect = lse35.login("", "password")
    print(f"Success: {success}, Redirect: {redirect}")
    
    print("\\nTest 5 - None inputs:")
    success, redirect = lse35.login(None, None)
    print(f"Success: {success}, Redirect: {redirect}")


if __name__ == "__main__":
    main()

File: benchmarkDS/ZeroShot/CLAUDE_claude-sonnet-4-5-20250929/Lse35.java

import java.sql.*;
import java.security.MessageDigest;
import java.security.NoSuchAlgorithmException;
import java.security.SecureRandom;
import java.util.Base64;

public class Lse35 {
    private static final String DB_URL = "jdbc:mysql://localhost:3306/userdb";
    private static final String DB_USER = "root";
    private static final String DB_PASSWORD = "password";
    
    public static class LoginResult {
        public boolean success;
        public String redirectUrl;
        
        public LoginResult(boolean success, String redirectUrl) {
            this.success = success;
            this.redirectUrl = redirectUrl;
        }
    }
    
    private static String hashPassword(String password, String salt) {
        try {
            MessageDigest md = MessageDigest.getInstance("SHA-256");
            md.update(Base64.getDecoder().decode(salt));
            byte[] hashedPassword = md.digest(password.getBytes());
            return Base64.getEncoder().encodeToString(hashedPassword);
        } catch (NoSuchAlgorithmException e) {
            throw new RuntimeException("Error hashing password", e);
        }
    }
    
    public static LoginResult login(String username, String password) {
        if (username == null || password == null || username.isEmpty() || password.isEmpty()) {
            return new LoginResult(false, "login.html");
        }
        
        Connection conn = null;
        PreparedStatement pstmt = null;
        ResultSet rs = null;
        
        try {
            conn = DriverManager.getConnection(DB_URL, DB_USER, DB_PASSWORD);
            String query = "SELECT password_hash, salt FROM users WHERE username = ?";
            pstmt = conn.prepareStatement(query);
            pstmt.setString(1, username);
            rs = pstmt.executeQuery();
            
            if (rs.next()) {
                String storedHash = rs.getString("password_hash");
                String salt = rs.getString("salt");
                String inputHash = hashPassword(password, salt);
                
                if (storedHash.equals(inputHash)) {
                    return new LoginResult(true, "home.html");
                }
            }
            return new LoginResult(false, "login.html");
            
        } catch (SQLException e) {
            e.printStackTrace();
            return new LoginResult(false, "login.html");
        } finally {
            try {
                if (rs != null) rs.close();
                if (pstmt != null) pstmt.close();
                if (conn != null) conn.close();
            } catch (SQLException e) {
                e.printStackTrace();
            }
        }
    }
    
    public static void main(String[] args) {
        // Test cases
        System.out.println("Test 1 - Valid user:");
        LoginResult result1 = login("john_doe", "SecurePass123");
        System.out.println("Success: " + result1.success + ", Redirect: " + result1.redirectUrl);
        
        System.out.println("\\nTest 2 - Invalid password:");
        LoginResult result2 = login("john_doe", "WrongPass");
        System.out.println("Success: " + result2.success + ", Redirect: " + result2.redirectUrl);
        
        System.out.println("\\nTest 3 - Non-existent user:");
        LoginResult result3 = login("fake_user", "password");
        System.out.println("Success: " + result3.success + ", Redirect: " + result3.redirectUrl);
        
        System.out.println("\\nTest 4 - Empty username:");
        LoginResult result4 = login("", "password");
        System.out.println("Success: " + result4.success + ", Redirect: " + result4.redirectUrl);
        
        System.out.println("\\nTest 5 - Null inputs:");
        LoginResult result5 = login(null, null);
        System.out.println("Success: " + result5.success + ", Redirect: " + result5.redirectUrl);
    }
}
